### ISSUE TITLE
Migration continues to generate same column alters over and over

### ISSUE THREAD

**Issue by Tarig0** [dotnet/efcore#11174]
When creating a migration the engine generates the following

```c#
protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AlterColumn<string>(
                name: "Type",
                table: "PartAttribute",
                nullable: false,
                oldClrType: typeof(string));

            migrationBuilder.AlterColumn<string>(
                name: "Type",
                table: "ElectricalPartVoltageCodes",
                nullable: false,
                oldClrType: typeof(string));
            // Every Time ^
            migrationBuilder.CreateIndex(
                name: "IX_ElectricalPartTypes_LabelPrefix",
                table: "ElectricalPartTypes",
                column: "LabelPrefix",
                unique: true);
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropIndex(
                name: "IX_ElectricalPartTypes_LabelPrefix",
                table: "ElectricalPartTypes");
            //every time V
            migrationBuilder.AlterColumn<string>(
                name: "Type",
                table: "PartAttribute",
                nullable: false,
                oldClrType: typeof(string));

            migrationBuilder.AlterColumn<string>(
                name: "Type",
                table: "ElectricalPartVoltageCodes",
                nullable: false,
                oldClrType: typeof(string));
        }
```
expect
```
protected override void Up(MigrationBuilder migrationBuilder)
        {
             migrationBuilder.CreateIndex(
                name: "IX_ElectricalPartTypes_LabelPrefix",
                table: "ElectricalPartTypes",
                column: "LabelPrefix",
                unique: true);
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropIndex(
                name: "IX_ElectricalPartTypes_LabelPrefix",
                table: "ElectricalPartTypes");
        }
```
I'm not sure what relevant information I need to gather

### Further technical details
EF Core version: 2.0.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Win 10
IDE: Visual Studio 2017 15.4

**Comment by ajcvickers** [comment]
@Tarig0 It's possible this is already fixed--could you try with 2.1 preview1 or our nightly builds? If not, can you please post a runnable project/solution or complete code listing that demonstrates the behavior you are seeing?

**Comment by Tarig0** [comment]
If I upgrade my migrations project alone,

module A 2.0.1
module B 2.0.1

migrations project 2.1.0

I get the following

```
add-migration -startupproject labels.migrations -project labels.migrations test
System.ArgumentNullException: Value cannot be null.
Parameter name: language
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations..ctor(IOperationReporter reporter, Assembly assembly, Assembly startupAssembly, String projectDir, String rootNamespace, String language)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.<>c__DisplayClass4_0.<.ctor>b__4()
   at Microsoft.EntityFrameworkCore.Internal.LazyRef`1.get_Value()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
```

**Comment by Tarig0** [comment]
Fixed in 2.1 can't wait for the official ready for prod release

**Comment by Tarig0** [comment]
To bad I can't just upgrade my migrations project.

migrations 2.1 -> extendmodel module a 2.0
   -> extendmodel module b 2.0

Looks like the error above comes in because the old 2.0 operations gets removed.
